src/modsim2/similarity/metrics/pad.py

```diff
@@ -0,0 +1,310 @@
+        """
+        # Split A and B into train and test datasets
+        # This is done before the data are concatenated in order
+        # to ensure that the same proportion of A & B are
+        # included in the test dataset
+        train_A, test_A, train_B, test_B = self.__train_test_split(data_A, data_B)
+
+        # concatenate A&B datasets
+        (
+            self.train_data,
+            self.train_labels,
+            self.test_data,
+            self.test_labels,
+        ) = self.__concat_data(train_A, test_A, train_B, test_B)
+
+        # Extract embedding callable
+        embedding_fn = EMBEDDING_FN_DICT[self.embedding_name]
+        # Transform the train and test data using the embedding function
+        embed_train_data = embedding_fn(self.train_data)
+        embed_test_data = embedding_fn(self.test_data)
+
+        return embed_train_data, embed_test_data
+
+    def __build_models(
+        self, kernel_name: str, c_values: list, gamma_values: list, degree_values: list
+    ):
+        """
+        This method will build the classifiers, variation in some hyperparameters
+        is allowed as the optimal values are not necessarily known for the datasets.
+        The kernel is not varied.
+        The trained model are appended to the object's list of models.
+
+        Args:
+            kernel_name:
+            c_values: list of C values (regularisation param) to be applied
+            gamma_values: list of gamma values (kernel coefficient) to be applied
+            degree_values: list of degree values for the poly kernel to be applied
+
+        Returns:
+            models: list of the models that have been built
+        """
+        models = []
+        for c in c_values:
+            for gamma in gamma_values:
+                for degree in degree_values:
+                    svc = SVC(kernel=kernel_name, C=c, gamma=gamma, degree=degree)
+                    svc.fit(X=self.embed_train_data, y=self.train_labels)
+                    models.append(svc)
+        return models
+
+    def __evaluate_models(self):
+        """
+        This method will evaluate the classifiers that have been built
+        The evaluation metric is the mean absolute error.
+
+        Returns:
+            errors: list of errors
+        """
+        errors = []
+        for svc in self.models:
+            preds = svc.predict(self.embed_test_data)
+            abs_diff = np.abs(preds - self.test_labels)
+            error = np.sum(abs_diff) / len(preds)
+            errors.append(error)
+        return errors
+
+    def calculate_distance(
+        self,
+        data_A: np.ndarray,
+        data_B: np.ndarray,
+        labels_A: np.ndarray,
+        labels_B: np.ndarray,
+        c_values: list,
+        kernel_name: str,
+        embedding_name: str,
+        test_proportion: float = 0.2,
+        train_proportion_equal=False,
+        gamma_values: list = ["scale"],
```

Comment:
Makes sense re the meaning of scale - could something be added to the docstring pointing either to the sklearn documentation or just explaining what it means as an option?

Reply:
I have added a reference to the sklearn docs